docs/LRM-feedback.md

```diff
@@ -0,0 +1,420 @@
+
+The scope of an identifier can be either global or local. A local identifier's scope is limited to inside the braces `{ ... }` in which they are declared, whereas global identifiers are declared outside any braces.
+
+## 4: Expressions
+
+#### 4.1 Primary Expressions
+
+Primary expressions in FIRE can either because accessing an array type or a function call. The primary expression is followed by an expression of either asignments in the case of a function call or an accessing of the datatypes in an array.
+
+1. `Primary Expression [ expression ]`
+2. `Primary Expression ( expression )`
+
+
+#### 4.2 Assignment Operator
+
+The assignment operators `=` returns the value of the expression that is evaluated on its right-hand side and stores it in the identifier on the left hand side. The scope of that identifier is described in [section 3](#Identifiers)
+
+#### 4.3 Function Calls
+
+Functions take in arguments by value except in the case of other functions which are passed by reference. Functions need to be assigned before being called but can be declared anonymously. An anonymous function's scope is within the function that it is declared.
+
+#### 4.4 Logical Negation
+
+TODO We're considering adding booleans, its easier to parse.
+
+Although FIRE does not directly support a boolean type, it equates `true` expressions to `1` and false expressions to `0`. As such, the logical negation operator `!` will convert the result of the logical expression on which it is applied to the inverse of what would normally be expected.
+
+All non-zero integers are evaluated as `true` where as `0` is false.
+
+#### 4.5 Logical AND Operator
+
+The logical AND `&&` is a short circuit operator, and returns 1 if and only if the expressions on its left and right both evaluate to 1.
+
+#### 4.6 Logical OR Operator
+
+The logical OR `||` operator is a short circuit operator and returns 1 if either of the expressions on its left or right return 1.
+
+#### 4.7 Relational Operators
+
+The relational operators `<, >, <=, <=, ==` return `1` if the expression on the left side of the operator has the expected relation to the operator on the right-hand side.
+
+These relationships amongst ints are determined by natural ordering. Strings can only be evaluated using the `==` operator.
+
+#### 4.8 Pattern Match Operator
+
+The pattern match operator `===` returns `1` if the regular expression or string on its right side conforms to the rules laid out by the regular expression on its left side and returns `0` otherwise.
+
+Pattern matching operator l-values must be type `regx` and r-values must be type `string`.
+
+Example: `r'[a-z]+' === word`
+
+#### 4.9 String Concatination Operator
+
+The string concatenation `^` operator returns a new string that is the concatenation of the string on its left side and the string on its right side.
+
+#### 4.10 Bracket Operator
+
+The bracket operator `[]` can be used with either`string` or `array`.
```

Comment:
by "used with", do you mean that the value within the brackets can be a string/array, or that it can operate on a string/array?

Reply:
You should specify the typing rules for the indexing operation. According to conventional understanding, the types for some operationg `a[i]` should be as follows:
```(a:array[k, v])[i:k] : v```
where the `k` and `v` are type variables. The syntax i'm using is `value:type`, that is the type comes after the `:`.

Reply:
> by "used with", do you mean that the value within the brackets can be a string/array, or that it can operate on a string/array?

yes

Reply:
> You should specify the typing rules for the indexing operation. According to conventional understanding, the types for some operationg `a[i]` should be as follows:
> `(a:array[k, v])[i:k] : v`
> where the `k` and `v` are type variables. The syntax i'm using is `value:type`, that is the type comes after the `:`.

Added-- thanks!